### PR TITLE
[dv,clkmgr] Generate clkmgr binds

### LINKED ISSUE #19448: [clkmgr] Make `clkmgr` DV top-level agnostic
### Description

When working on Integrated, we've noticed that clkmgr DV is not always top-level agnostic and in some cases is focused on the clkmgr instance of Earlgrey. For example, the top-level specific `clkmgr_bind` is hardcoded rather than being generated from a template. This is not ideal as it doesn't work for other top levels.

### PR DESCRIPTION
**Pull request by matutem**
Generate the sva and coverage binds so they can support multiple tops. The coverage and sva binds are specific to top_earlgrey.

Fixes #19448

### CODE REVIEW

File: hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv
Comment (vogelpi) [reply hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv:165]: But the clock gate enable is the same.

**Review by vogelpi — APPROVED** [review]
This looks mostly good @matutem , thanks for doing it!

I'v noticed some differences in the `hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv` file which is the one that really matters for the tapeout. Before merging this, can you please clarify whether the marked differences are intentional and if yes, explain why? Thanks!

File: hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv
Comment (matutem) [reply hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv:248]: It was missing in the previous code. This is why generation is better :)

File: hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv
Comment (matutem) [reply hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv:165]: Interestingly this change doesn't matter: both clk and rst_n ports of clkmgr_aon_cg_en_sva_if are unused, so this change is irrelevant. Same below. I removed the unused ports.

**Comment by matutem** [comment]
Good catch on the difference for clkmgr_aon_cg_en_sva_if. Removing the unused ports makes things simpler.

File: hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv
Comment (vogelpi) [reply hw/top_earlgrey/ip_autogen/clkmgr/dv/sva/clkmgr_bind.sv]: Ah, I see. Thanks for explaining and further simplifying this @matutem .

**Comment by vogelpi** [comment]
Thanks @matutem for your responses and further improvements. I'll merge this once CI passes.

**Pull request merged**
